### PR TITLE
feat: Performance log enhancement

### PR DESCRIPTION
**Pull request by Eswar2103**

### CODE REVIEW

**Review by neeradanelxsi — APPROVED** [review]

**Review by ibrahimshuail — APPROVED** [review]

File: cypress/support/step_definitions/validations.js [hunk 1]
```diff
@@ -298,17 +298,27 @@ Given(
...
-      } else {
-        result.map((response) => {
-          cy.log(response.message).then(() => {
-            assert.equal(true, response?.success, response?.message);
+      fireLog.info('Performance validation has stopped').then(() => {
+        cy.wrap(result)
+          .then((response) => {
+            if (response && Array.isArray(response)) {
+              response.map((res) => {
+                cy.log(JSON.stringify(res));
```
Comment (jbigel) [inline cypress/support/step_definitions/validations.js:306]: Can we use fireLog?

File: cypress/support/step_definitions/validations.js [hunk 2]
```diff
@@ -298,17 +298,27 @@ Given(
...
+              });
+            }
+          })
+          .then(() => {
+            if (result.error) {
+              cy.log('Failed to fetch and validate the performance metrics').then(() => {
```
Comment (jbigel) [inline cypress/support/step_definitions/validations.js:312]: Can we use fireLog?

File: cypress/support/step_definitions/validations.js [hunk 3]
```diff
@@ -298,17 +298,27 @@ Given(
...
+                assert(false, result.error);
+              });
+            } else {
+              result.map((response) => {
+                assert.equal(true, response?.success, response?.message);
```
Comment (jbigel) [inline cypress/support/step_definitions/validations.js:317]: Can we use fireLog?

File: cypress/support/validations/performanceValidation.js
Comment (jbigel) [inline cypress/support/validations/performanceValidation.js:1]: Same questions as above

Comment (PreethiMaai) [reply cypress/support/validations/performanceValidation.js:1]: updated

File: cypress/support/step_definitions/validations.js [hunk 4]
```diff
@@ -298,27 +298,25 @@ Given(
...
+        .then((response) => {
+          if (response && Array.isArray(response)) {
+            response.map((res) => {
+              fireLog.info(JSON.stringify(res));
+            });
+          }
+        })
+        .then(() => {
+          if (result.error) {
+            fireLog.info('Failed to fetch and validate the performance metrics').then(() => {
```
Comment (jbigel) [inline cypress/support/step_definitions/validations.js:311]: I think Joe created a fireLog command that can be used that handles the `fireLog.info() -> fireLog.assert()` sequence that you have here. Please check it out: https://github.com/rdkcentral/firebolt-certification-suite/blob/dev/cypress/support/cypress-support/src/utils.js#L882 and use it if it handles this case.

File: cypress/support/validations/performanceValidation.js [hunk 5]
```diff
@@ -47,18 +47,18 @@ Cypress.Commands.add('performanceValidation', (object) => {
...
-              cy.log(JSON.stringify(res));
+              fireLog.info(JSON.stringify(res));
             });
           }
         })
         .then(() => {
           if (result.error) {
-            cy.log('Failed to fetch and validate the performance metrics').then(() => {
-              assert(false, result.error);
+            fireLog.info('Failed to fetch and validate the performance metrics').then(() => {
```
Comment (jbigel) [inline cypress/support/validations/performanceValidation.js:56]: same as above

**Review by jbigel — APPROVED** [review]

File: cypress/support/step_definitions/validations.js (hunk 4 above)
Comment (PreethiMaai) [reply cypress/support/step_definitions/validations.js:311]: I tried using firelog.error, but didn't see any difference in the logs, so I kept it as is. Here are the reports for [firelog.assert]( https://reporting.firecert.comcast.com/cucumberReports/bd10f4d3-e219-4cc3-a4cc-c8b0c9ef15f4/index.html
) and [firelog.error](https://reporting.firecert.comcast.com/cucumberReports/a5644971-494d-4579-bf66-d909bf767823/index.html)

**Pull request merged**